lib/periodicChecks.js

```diff
@@ -0,0 +1,213 @@
+const {
+  getAllOpenPullRequests,
+  hasPullRequestBeenApproved,
+  getChangelogLabelFromPullRequest,
+  getProjectOwnerFromLabel,
+  doesPullRequestHaveChangesRequested,
+} = require('./utils');
+const mergeConflictModule = require('./checkMergeConflicts');
+const { teamLeads } = require('../userWhitelist.json');
+
+/**
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasMergeConflict = (pullRequest) => {
+  return pullRequest.mergeable === false;
+};
+
+/**
+ *
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const hasPendingReviews = (pullRequest) => {
+  return pullRequest.requested_reviewers.length > 0;
+};
+
+/**
+ * This function pings and assigns all pending reviewers to a pull request.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const pingAndAssignReviewers = async (context, pullRequest) => {
+  const reviewers = pullRequest.requested_reviewers.map(
+    (reviewer) => reviewer.login
+  );
+
+  context.github.issues.addAssignees(
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: reviewers,
+    })
+  );
+
+  context.github.issues.createComment(
+    context.repo({
+      issue_number: pullRequest.number,
+      body:
+        'Assigning @' +
+        reviewers.join(', @') +
+        ' for codeowner reviews. Thanks!',
+    })
+  );
+};
+
+/**
+ * This function assigns the PR author to a pull request.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const assignPrAuthor = async (context, pullRequest) => {
+  context.github.issues.addAssignees(
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: [pullRequest.user.login],
+    })
+  );
+};
+
+/**
+ * This function handles when a pull request has been approved.
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const handleApproval = async (context, pullRequest) => {
```

Comment:
I am a bit confused - what is the purpose here - don't we have a similar method in #116 as well?